tests/test_all_rules.py

```diff
@@ -987,22 +987,31 @@ def test_event_override(self):
         errors = []
 
         for rule in self.all_rules:
-            # skip rules that do not leverage queries (i.e. machine learning)
-            # filters to acceptable query languages in definitions.FilterLanguages
-            # QueryRuleData should inherently ignore machine learning rules
-            if isinstance(rule.contents.data, QueryRuleData):
-                rule_language = rule.contents.data.language
-                has_event_ingested = rule.contents.data.get('timestamp_override') == 'event.ingested'
-                rule_str = self.rule_str(rule, trailer=None)
-
-                if not has_event_ingested:
-                    # TODO: determine if we expand this to ES|QL
-                    # ignores any rule that does not use EQL or KQL queries specifically
-                    # this does not avoid rule types where variants of KQL are used (e.g. new terms)
-                    if rule_language not in ('eql', 'kuery') or getattr(rule.contents.data, 'is_sequence', False):
-                        continue
-                    else:
-                        errors.append(f'{rule_str} - rule must have `timestamp_override: event.ingested`')
+            # Skip rules that do not leverage queries (i.e., machine learning)
```

Comment:
So this won't actually address the root cause of the issue here. This test ensures we _are_ setting the `timestamp_override`, however, it does not check that it is available in the data schema.

One thing we may be able to do is during the validation process, we check

```python
if data.timestamp_override:
    # check the schema which is already built and look for `event.ingested` or whatever the value is
```

Reply:
Great point here, will update unit test.

Reply:
What is interesting is that aside from the ML ones, AuditD Manager, Sentinel One Cloud Funnel and Cloud Defend have rules that leverage `event.ingested` but the field is not found in the respective integrations 🤔 

I see `event.ingested` in alert telemetry, so I wonder if there is a default? I was unable to locate `event.ingested` anywhere in the `auditd_manager` integration. [Reference](https://github.com/search?q=repo%3Aelastic%2Fintegrations+path%3A%2F%5Epackages%5C%2Fauditd_manager%5C%2F%2F+%22event.ingested%22&type=code).

Reply:
I've posted a question in security integrations, regarding the others. Unfortunately, this would be a blocker if all other integrations were not fixed and indeed do not have `event.ingested` -> https://elastic.slack.com/archives/C010LT16CFR/p1716479083311779